src/Groups/types.jl

```diff
@@ -432,6 +445,10 @@ mutable struct MatrixGroupElem{RE<:RingElem, T<:MatElem{RE}} <: AbstractMatrixGr
    end
 end
 
+function Base.hash(x::MatrixGroupElem, h::UInt)
+  return hash(matrix(x), hash(base_ring(parent(x)), h))
```

Comment:
```suggestion
  return hash(matrix(x), hash(parent(x), h))
```
IMO it is better to compose hash methods, ie just take the hash of the parent here, and then add a hash function for the parent that takes the base_ring into account

Reply:
Actually, at least at this time, we should take neither the parent nor the base ring into account, because the `==` also doesn't:
```julia
# we are not currently keeping track of the parent; two elements coincide iff their matrices coincide
function ==(x::MatrixGroupElem{S,T},y::MatrixGroupElem{S,T}) where {S,T}
   if isdefined(x,:X) && isdefined(y,:X) return x.X==y.X
   else return matrix(x)==matrix(y)
   end
end
```

This method is defined in `src/Groups/matrices/MatGrp.jl:449` -- I think it would be a good idea to put the `==` and `hash` method next to each other, given that they need to be in sync.

Reply:
```suggestion
  return hash(matrix(x), h)
```